app/request-actions/private-cloud/edit-request.ts

```diff
@@ -52,19 +52,32 @@ export default async function editRequest(
       : undefined,
   };
 
-  // The edit request will require manual admin approval if any of the quotas are being changed.
-  const isNoQuotaChanged =
-    JSON.stringify(formData.productionQuota) === JSON.stringify(project.productionQuota) &&
-    JSON.stringify(formData.testQuota) === JSON.stringify(project.testQuota) &&
-    JSON.stringify(formData.developmentQuota) === JSON.stringify(project.developmentQuota) &&
-    JSON.stringify(formData.toolsQuota) === JSON.stringify(project.toolsQuota);
-
   let decisionStatus: DecisionStatus;
 
   const hasGolddrEnabledChanged = project.cluster === Cluster.GOLD && project.golddrEnabled !== formData.golddrEnabled;
 
+  const requestedQuota = {
+    testQuota: formData.testQuota,
+    toolsQuota: formData.toolsQuota,
+    developmentQuota: formData.developmentQuota,
+    productionQuota: formData.productionQuota,
+  };
+
+  const currentQuota = {
+    testQuota: project.testQuota,
+    toolsQuota: project.toolsQuota,
+    developmentQuota: project.developmentQuota,
+    productionQuota: project.productionQuota,
+  };
+
+  const noQuotaChange = checkIfNoQuotaChange(currentQuota, requestedQuota);
   // If there is no quota change or no quota upgrade and no golddr flag changes, the request is automatically approved
-  if ((isNoQuotaChanged || !isQuotaUpgrade(formData, project)) && !hasGolddrEnabledChanged) {
+  if (noQuotaChange) {
+    decisionStatus = DecisionStatus.APPROVED;
+  } else if (
+    checkIfQuotaAutoApproval(currentQuota, requestedQuota, project.licencePlate, project.cluster) &&
```

Comment:
put `await` front of `checkIfQuotaAutoApproval(currentQuota, requestedQuota, project.licencePlate, project.cluster)`